torch/distributed/distributed_c10d.py

```diff
@@ -1111,6 +1111,37 @@ def _check_tensor_list(param, param_name) -> None:
         )
 
 
+def _group_or_default_group(group: Optional[ProcessGroup] = None) -> ProcessGroup:
+    if group is None or group is GroupMember.WORLD:
+        group = _get_default_group()
+    return group
+
+
+def _canonicalize_group_rank(
+    group: ProcessGroup,
+    global_rank: Optional[int] = None,
+    group_rank: Optional[int] = None,
+) -> int:
+    """
+    Helper method to take _either_ a global rank or a group rank and produce a group rank.
+    """
+
+    if group_rank is not None:
+        assert global_rank is None, "Can't specify both group_rank and global_rank"
+    else:
+        assert global_rank is not None, "Must specify global_rank or group_rank"
+        group_rank = get_group_rank(group, global_rank)
+    return group_rank
+
+
+def _check_not_self_rank(group: ProcessGroup, rank: int, rank_type: str):
+    if group.rank() == rank:
+        raise ValueError(
+            f"Invalid {rank_type} rank: {rank_type} rank should not be the same as "
```

Comment:
Did you mean to print `rank_type` twice here?

Reply:
Yea it's repetitive but I was preserving the original log message